controllers/k8ssandra/auth_test.go

```diff
@@ -127,8 +127,6 @@ func createSingleDcClusterNoAuth(t *testing.T, ctx context.Context, f *framework
 	err = f.DeleteK8ssandraCluster(ctx, client.ObjectKey{Namespace: kc.Namespace, Name: kc.Name}, timeout, interval)
 	require.NoError(t, err, "failed to delete K8ssandraCluster")
 	f.AssertObjectDoesNotExist(ctx, t, dcKey, &cassdcapi.CassandraDatacenter{}, timeout, interval)
-	f.AssertObjectDoesNotExist(ctx, t, stargateKey, &stargateapi.Stargate{}, timeout, interval)
```

Comment:
Owner references are not actually cleaned up in env tests, so we verify the presence of the reference instead of the deletion. See also the comment on `IsOwnedByCassandraDatacenter`.

Reply:
Ah, I see. OK, so a test for this functionality wouldn't work anyway because OwnerReferences don't result in a cleanup? But because this was previously manually cleaned up it used to work? 

I'm OK to resolve this comment if so - it makes sense to me. I'd forgotten how limited the envests were.